crates/turbopack-bench/src/lib.rs

```diff
@@ -0,0 +1,497 @@
+use std::{
+    fs::{self},
+    panic::AssertUnwindSafe,
+    path::Path,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use anyhow::{anyhow, Context, Result};
+use criterion::{
+    measurement::{Measurement, WallTime},
+    BenchmarkGroup, BenchmarkId, Criterion,
+};
+use once_cell::sync::Lazy;
+use tokio::{
+    runtime::Runtime,
+    time::{sleep, timeout},
+};
+use turbo_tasks::util::FormatDuration;
+use util::{
+    build_test, create_browser,
+    env::{read_env, read_env_bool, read_env_list},
+    module_picker::ModulePicker,
+    AsyncBencherExtension, PreparedApp, BINDING_NAME,
+};
+
+use self::{bundlers::RenderType, util::resume_on_error};
+use crate::{bundlers::Bundler, util::PageGuard};
+
+pub mod bundlers;
+pub mod util;
+
+const MAX_UPDATE_TIMEOUT: Duration = Duration::from_secs(60);
+
+pub fn bench_startup(c: &mut Criterion, bundlers: &[Box<dyn Bundler>]) {
```

Comment:
a `bundlers` argument has been added to all these methods to call them from the actual benchmark with a list of bundlers